### DISCUSSION TITLE
How to deserialize vectors?

### DISCUSSION THREAD

**Discussion by reigenatk** [nlohmann/json#2463] [Q&A] [answered]
Hello, does this library provide a way to deserialize vectors from a JSON file? 

What I mean is if I have something like this in a json file
```cpp
{
    "v": {3, 2, 3}
}
```

Then I'm trying to do this:

```cpp
std::ifstream i("pathtojsonfile");
json j;
i >> j;
vector<int> v = j["test"]
```

But this isn't working. So how then do I do it? I looked through the docs carefully but don't see anything about deserializing into a vector, rather I only see stuff about serializing. Thanks!

**Comment by nlohmann** [comment, marked as answer]
The example above is not valid JSON. It should be

```json
{
    "v": [3, 2, 3]
}
```

Then you can access the array with `j["v"]`.
